### PR TITLE
Add logging metrics

### PR DESCRIPTION
**Pull request by kuba-krj**
# Description
Add logging loss/flops and loss/tokens

# Checklist
   - [x] Is this PR focused on a single feature without the possibility of being divided into smaller PRs?
   - [x] Are all variables set with logical defaults that are backward compatible?
   - [x] I attached link to neptune if it was necessary [LLMRANDOM-3891](https://app.neptune.ai/o/pmtest/org/llm-random/runs/details?viewId=9aaf02e4-ed95-4ef1-9fe3-1638b70c08ac&shortId=LLMRANDOM-3891&type=run&compare=MwDgnAjANBQ)
   - [x] Have you successfully executed the linter and tests?
   - [x] Are all functions concise and don't require splitting?
   - [x] Is there documentation for arguments and complex logic?
   - [x] Are there no temporary solutions in the code? Should a task be added to Trello to refine the code?
   - [x] Are all functions placed in appropriate files?
   - [x] Is the PR name meaningful and descriptive?
   - [x] Is PR description provided, or unnecessary?
  
Other questions:
   - [x] If I made changes to the requirements or anything related to the Singularity image I did generate new image and upload a new image to the clusters
   - [x] If this change is significant enough I notified all individuals working with this repository (@channel in slack channel)
   - [x] If there is a need for a second reviewer I requested additional review 

# Reviewer's checklist:
   - [ ] Does the Neptune link work and does it actually compare the situation before and after the PR?
   - [ ] Do I understand the code?
   - [ ] Do I think another reviewer is not needed or I commented that this PR need another reviewer?

### CODE REVIEW

**Review by ludziej — CHANGES_REQUESTED** [review]
some changes discussed on PM

File: research/conditional/utils/conditional_trainer.py [hunk 1]
```diff
@@ -336,6 +337,16 @@ def _log_train_stats(self, loss_value, step):
...
+                self.logger.report_scalar(
+                    title=f"{name}_flops",
+                    value=stats.acc / stats.interval,
+                    iteration=step * self.flops_per_step,
+                )
+                self.logger.report_scalar(
+                    title=f"{name}_tokens",
+                    value=stats.acc / stats.interval,
+                    iteration=step * self.batch_size * self.max_sequence_length,
+                )
```
Comment (crewtool) [inline research/conditional/utils/conditional_trainer.py:349]: Are you sure this is correct? Shouldn't 'iteration' be a step? Like this?
```suggestion
                self.logger.report_scalar(
                    title=f"{name}_flops",
                    value=step * self.flops_per_step,
                    iteration=step,
                )
                self.logger.report_scalar(
                    title=f"{name}_tokens",
                    value=step * self.batch_size * self.max_sequence_length
                    iteration=step,
                )
```
I've got the point to measure how much we actually processed achieving given loss, but not sure it is possible this way

**Comment by crewtool** [comment]
I have found some FB library for flops calculation, did you see it? https://github.com/facebookresearch/fvcore/blob/main/docs/flop_count.md

**Comment by ludziej** [comment]
> I have found some FB library for flops calculation, did you see it? https://github.com/facebookresearch/fvcore/blob/main/docs/flop_count.md

This looks interesting, but the point in this PR was to compare our computational complexity and theoretical FLOPS calculation with actual training time. 

Library you've mentioned may be a good idea for another PR.

File: research/conditional/utils/conditional_trainer.py (hunk 1 above)
Comment (ludziej) [reply research/conditional/utils/conditional_trainer.py:349]: I think that's correct and does exactly what we wanted to: produce a loss/tokens or a loss/flops plot.

**Review by Simontwice — CHANGES_REQUESTED** [review]
Please make the logging of this optional with an argparse flag

**Comment by maciek-pioro** [comment]
@kuba-krj @ludziej is this still relevant?

**Comment by crewtool** [comment]
I am closing it due to no response.

**Pull request closed without merging**